apps/webClient/src/Model/Threejs/SceneManager.js

```diff
@@ -0,0 +1,109 @@
+import * as THREE from "three";
+import SceneSubject from "./SceneSubject";
+import GeneralLights from "./GeneralLights";
+
+export default canvas => {
+  const clock = new THREE.Clock();
+  const origin = new THREE.Vector3(0, 0, 0);
+
+  const screenDimensions = {
+    width: canvas.width,
+    height: canvas.height
+  };
+
+  const mousePosition = {
+    x: 0,
+    y: 0
+  };
+
+  const scene = buildScene();
+  const renderer = buildRender(screenDimensions);
+  const camera = buildCamera(screenDimensions);
+  const sceneSubjects = createSceneSubjects(scene);
+
+  function buildScene() {
+    const scene = new THREE.Scene();
+    scene.background = new THREE.Color("#FFF");
+
+    return scene;
+  }
+
+  function buildRender({ width, height }) {
+    const renderer = new THREE.WebGLRenderer({
+      canvas: canvas,
+      antialias: true,
+      alpha: true
+    });
+    const DPR = window.devicePixelRatio ? window.devicePixelRatio : 1;
+    renderer.setPixelRatio(DPR);
+    renderer.setSize(width, height);
+
+    renderer.gammaInput = true;
+    renderer.gammaOutput = true;
+
+    return renderer;
+  }
+
+  function buildCamera({ width, height }) {
+    const aspectRatio = width / height;
+    const fieldOfView = 60;
+    const nearPlane = 4;
+    const farPlane = 100;
+    const camera = new THREE.PerspectiveCamera(
+      fieldOfView,
+      aspectRatio,
+      nearPlane,
+      farPlane
+    );
+
+    camera.position.z = 40;
+
+    return camera;
+  }
+
+  function createSceneSubjects(scene) {
+    const sceneSubjects = [new GeneralLights(scene), new SceneSubject(scene)];
+
+    return sceneSubjects;
+  }
+
+  function update() {
+    const elapsedTime = clock.getElapsedTime();
+
+    for (let i = 0; i < sceneSubjects.length; i++)
+      sceneSubjects[i].update(elapsedTime);
+
+    updateCameraPositionRelativeToMouse();
+
+    renderer.render(scene, camera);
+  }
```

Comment:
curious why [requestAnimationFrame](https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame) isn't used here -- I see it in most update examples